src/web3/EventPoller.ts

```diff
@@ -26,9 +29,47 @@ export class EventPoller extends EventEmitter {
 
 	private async callback() {
 		try {
-			const events: Array<IEvent> = await this.web3Client.smartContracts().getFilteredScOutputEvents(this.eventsFilter);
-            if (events.length > 0) {
-                this.emit(ON_MASSA_EVENT_DATA, events);
+            // get all events using the filter
+			const events: Array<IEvent> = await this.web3Client
+			.smartContracts()
+			.getFilteredScOutputEvents(this.eventsFilter);
+
+            // filter further using regex and last scanned slot
+            const filteredEvents: Array<IEvent> = events.filter(event => {
+
+				// check if regex condition is met
+                let meetsRegex = true;
+				if ((this.eventsFilter as IEventRegexFilter).eventsNameRegex) {
+					meetsRegex = event.data.includes((this.eventsFilter as IEventRegexFilter).eventsNameRegex);
+				}
+                
+				// check if after last slot
+                let isAfterLastSlot = true;
+                if (this.lastSlot) {
+                    isAfterLastSlot = (event.context.slot.period > this.lastSlot.period) 
+					&& (event.context.slot.thread >= this.lastSlot.thread);
+                }
+				
+                return meetsRegex && isAfterLastSlot;
+            });
+
+			// sort after highest period and thread
+			const sortedByHighestThread = filteredEvents.sort((a, b) => {
```

Comment:
I believe you want to sort so that you obtain the latest slot. The latest slot is the one with the biggest period. If there are more than one slot with the biggest period, it is the slot with the biggest thread among those.
I believe you did the opposite (thread then period).

Reply:
@qdrn Thanks for the comment. I have fixed that. Can you please do another review round ?